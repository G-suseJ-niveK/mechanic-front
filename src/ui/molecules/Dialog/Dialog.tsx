import React from 'react';
import { Dialog, Divider, DialogActions, DialogContent, DialogTitle, Box } from '@material-ui/core';

type DialogProps = {
  actions?: React.ReactNode;
  children?: React.ReactNode;
  onClose: (value: boolean) => void;
  open: boolean;
  title?: string;
  subtitle?: string;
};

const ComponentDialog: React.FC<DialogProps> = (props: DialogProps) => {
  const { open, title, subtitle, onClose, children, actions } = props;

  const handleClose = () => {
    onClose(false);
  };

  return (
    <Dialog
      // keepMounted
      fullWidth
      open={open}
      onClose={handleClose}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
      max-width="500px">
      {title && (
        <DialogTitle>
          <Box display="flex" justifyContent="center">
            <span> {title} </span>
          </Box>
          <Box display="flex" justifyContent="center" color="#3D4C63" fontSize="0.9rem">
            <span> {subtitle} </span>
          </Box>
        </DialogTitle>
      )}
      <Divider />
      <DialogContent>{children}</DialogContent>
      <DialogActions>{actions}</DialogActions>
    </Dialog>
  );
};

export default React.memo(ComponentDialog);
