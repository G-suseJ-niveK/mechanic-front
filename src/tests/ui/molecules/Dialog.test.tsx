import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Dialog from '~molecules/Dialog/Dialog';

describe('Render Dialog', () => {
  const title: string = 'Test title';
  const actions: React.ReactNode = <div>actions</div>;
  const children: React.ReactNode = <div>children</div>;

  test('should no render', () => {
    const { container } = render(<Dialog onClose={() => null} open={false} />);
    expect(container).toMatchSnapshot();
  });

  test('should render title', () => {
    render(<Dialog onClose={() => null} open={true} title={title} />);
    expect(screen.getByText(title)).toBeInstanceOf(Node);
  });

  test('should render actions', () => {
    render(<Dialog onClose={() => null} open={true} actions={actions} />);
    expect(screen.getByText('actions')).toBeInstanceOf(Node);
  });

  test('should render children', () => {
    render(
      <Dialog onClose={() => null} open={true}>
        {children}
      </Dialog>
    );
    expect(screen.getByText('children')).toBeInstanceOf(Node);
  });

  test('should open and close', () => {
    let isOpen: boolean = true;

    const handleOnClose = jest.fn(() => {
      isOpen = false;
    });

    const button: React.ReactNode = <button onClick={handleOnClose}> click </button>;

    const { rerender } = render(<Dialog onClose={handleOnClose} open={isOpen} actions={button} />);
    expect(screen.getByRole('dialog')).toBeInstanceOf(Node);
    fireEvent.click(screen.getByText(/click/i));
    expect(isOpen).toBe(false);
    rerender(<Dialog onClose={handleOnClose} open={isOpen} actions={button} />);
    expect(screen.getByRole('dialog')).toBeInstanceOf(Node);
    // expect(container.firstChild).toBeNull();
  });
});
