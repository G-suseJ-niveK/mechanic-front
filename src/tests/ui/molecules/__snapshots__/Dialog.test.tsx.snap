// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Dialog should no render 1`] = `<div />`;
